docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+file = CSV.File(http_response)
+```
+
+#### [Reading from a zip file](@id zip_example)
+
+```julia
+using ZipFile, CSV, DataFrames
+
+a = DataFrame(a = 1:3)
+CSV.write("a.csv", a)
+
+# zip the file; Windows users who do not have zip available on the PATH can manually zip the CSV
+# or write directly into the zip archive as shown below
+;zip a.zip a.csv
+
+# alternatively, write directly into the zip archive (without creating an unzipped csv file first)
+z = ZipFile.Writer("a2.zip")
+f = ZipFile.addfile(z, "a.csv", method=ZipFile.Deflate)
+a |> CSV.write(f)
+close(z)
+
+# read file from zip archive
+z = ZipFile.Reader("a.zip") # or "a2.zip"
+
+# identify the right file in zip
+a_file_in_zip = filter(x->x.name == "a.csv", z.files)[1]
+
+a_copy = CSV.File(read(a_file_in_zip)) |> DataFrame
+
+a == a_copy
+```
+
+### [Column names on 2nd row](@id second_row_header)
+
+```julia
+using CSV
+
+data = """
+descriptive row with information about the file that we'd like to ignore
+a,b,c
+1,2,3
+4,5,6
+"""
+
+# by passing header=2, parsing will ignore the 1st row entirely
+# then parse the column names on row 2, then by default, it assumes
+# the data starts on the row after the column names (row 3 in this case)
+# which is correct for this case
+file = CSV.File(IOBuffer(data); header=2)
+```
+
+### [No column names in data](@id no_header)
+
+```julia
+using CSV
+
+# in this case, our data doesn't have any column names
+data = """
+1,2,3
+4,5,6
+"""
+
+# by passing `header=false`, parsing won't worry about looking for column names
+# anywhere, but instead just start parsing the data and generate column names
+# as needed, like `Column1`, `Column2`, and `Column3` in this case
+file = CSV.File(IOBuffer(data); header=false)
+```
+
+### [Manually provide column names](@id manual_header)
+
+```julia
+using CSV
+
+# in this case, our data doesn't have any column names
+data = """
+1,2,3
+4,5,6
+"""
+
+# instead of passing `header=false` and getting auto-generated column names,
+# we can instead pass the column names ourselves
+file = CSV.File(IOBuffer(data); header=["a", "b", "c"])
+
+# we can also pass the column names as Symbols; a copy of the manually provided
+# column names will always be made and then converted to `Vector{Symbol}`
+file = CSV.File(IOBuffer(data); header=[:a, :b, :c])
+```
+
+### [Multi-row column names](@id multi_row_header)
+
+```julia
+using CSV
+
+# in this case, our column names are `col_a`, `col_b`, and `col_c`,
+# but split over the first and second rows
+data = """
+col,col,col
+a,b,c
+1,2,3
+4,5,6
+"""
+
+# by passing a collection of integers, parsing will parse each row in the collection
+# and concatenate the values for each column, separating rows with `_` character
+file = CSV.File(IOBuffer(data); header=[1, 2])
+```
+
+### [Normalizing column names](@id normalize_header)
+
+```julia
+using CSV
+
+# in this case, our data are single letters, with column names of "1", "2", and "3"
```

Comment:
maybe add a comment that normalizing is not required but it is optional.

Reply:
added